maggy/core/environment/base.py

```diff
@@ -0,0 +1,186 @@
+#
+#   Copyright 2021 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import os
+import shutil
+import warnings
+
+from maggy import util
+
+class BaseEnv:
+    """
+    Support maggy on a local pyspark installation.
+    """
+    def __init__(self):
+        self.log_dir = os.path.join(os.getcwd(),'experiment_log')
+        if not os.path.exists(self.log_dir):
+            os.mkdir(self.log_dir)
+
+    def set_ml_id(self, app_id = 0, run_id = 0):
+        os.environ['ML_ID'] = str(app_id) + '_' + str(run_id)
+
+    def create_experiment_dir(self, app_id, run_id):
+        if not os.path.exists(os.path.join(self.log_dir, app_id)):
+            os.mkdir(os.path.join(self.log_dir, app_id))
+
+        experiment_path = self.get_logdir(app_id, run_id)
+        if os.path.exists(experiment_path):
+            shutil.rmtree(experiment_path)
+
+        os.mkdir(experiment_path)
+
+    def get_logdir(self, app_id, run_id):
+        return os.path.join(self.log_dir, str(app_id), str(run_id))
+
+    def populate_experiment(self, model_name, function, type, hp, description, app_id, direction, optimization_key):
+        pass
```

Comment:
We have to be careful with these functions in the base environment. The way Maggy works atm returning None could crash the experiment since it's relying on the function to return a dictionary (not 100% sure if it looks up specific keys). I don't think that's something to be fixed in this PR, but rather has to be changed by making all logging etc in the experiment optional.